### ISSUE TITLE
Change breaks behaviour of resource API description

### ISSUE THREAD

**Issue by jhiemer** [patrickcrocker/cf-cli-resource#31]
Hi,
due to the change here: https://github.com/patrickcrocker/cf-cli-resource/blob/master/assets/cf-functions.sh#L17

It is not possible to *not* provide org and space, although it is described like that in the readme.md. What is the reason for this change?

**Comment by patrickcrocker** [comment]
This function library is [pulled in](https://github.com/patrickcrocker/cf-cli-resource/blob/500bf9c9b7d50109c41d940913c1956467c0d368/assets/out#L10) by the `out` script which sets [strict checking](https://github.com/patrickcrocker/cf-cli-resource/blob/500bf9c9b7d50109c41d940913c1956467c0d368/assets/out#L4) for unset variables.

So, even the previous behavior would fail with an ambiguous error message:
```
cf-functions.sh: line 19: $1: unbound variable
```

The recent change doesn't change that behavior, but rather puts out a more clear message about the parameter that is missing:
```
cf-functions.sh: line 17: 1: org null or not set
```

The `org` is optional at the resource definition, but required for any command that targets an `org`.

What command are you trying to run that you do not have an org configured?

**Comment by jhiemer** [comment]
@patrickcrocker I am pretty sure its has been running two days, because we executing from a concourse pipeline. 

What we are doing is creating service brokers, where you don't need to set a org/space.

What I did was:

```yaml
resources:
- (( prepend ))
- name: cf-env
  type: cf-cli-resource
  source:
    api: (( grab config.api )) 
    username: (( grab config.username ))
    password: (( grab config.password ))
    skip_cert_check: (( grab config.skip_cert_check ))
```

```yaml
- put: cf-create-service-broker
      resource: cf-env
      params:
        command: create-service-broker
        service_broker: (( grab config.service_broker.name ))
        username: (( grab config.service_broker.username ))
        password: (( grab config.service_broker.password ))
        url: (( concat "https://" config.service_broker.host "." config.cf-dev.domain ))
```

**Comment by patrickcrocker** [comment]
Fixed in 2.5.1/latest.

org and space are now only required when setting the service broker as `space_scoped`.

Thanks for reporting this!
